string str
---
float64[] weights